packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -116,6 +116,14 @@ export interface RestApiProps extends ResourceOptions {
    */
   readonly endpointTypes?: EndpointType[];
 
+  /**
+   * A list of the VPC Endpoint IDs of the API. Use this property when creating
+   * a private API.
+   *
+   * @default - No VPC Endpoint IDs.
+   */
+  readonly vpcEndpointIds?: string[];
```

Comment:
I believe this should be of type `IVpcEndpoint` from the `@aws-cdk/aws-ec2` module.

Reply:
Updated to use VPCEndpoint. This adds dependency between api gateway and ec2. I moved the dependency from dev dependency to a real dependency with peer requirement.